nixos/modules/hardware/video/high-dpi.nix

```diff
@@ -0,0 +1,16 @@
+{ lib, pkgs, config, ...}:
+with lib;
+
+{
+  options.hardware.video.high-dpi.enable = mkEnableOption "Font/DPI configuration optimized for High-DPI displays";
+
+  config = mkIf config.hardware.video.high-dpi.enable {
+    console.font = lib.mkDefault "latarcyrheb-sun32";
+
+    # Needed when typing in passwords for full disk encryption
+    console.earlySetup = mkDefault true;
+    boot.loader.systemd-boot.consoleMode = mkDefault "1";
```

Comment:
I tried this on a Huawei MateBook X Pro 2018. Unfortunately it resulted in a small bootloader interface centred in the middle of the screen rather than stretching to fill the entire thing, but the default isn't great either; hopefully this will help on some hardware.

I have a minor worry that if changing this from the default of `"keep"` ends up changing the actual display resolution, it'll counteract the general ecosystem efforts towards keeping the native display mode throughout the boot process to achieve a flicker-free boot (see e.g. [Fedora Project Wiki](https://fedoraproject.org/wiki/Changes/FlickerFreeBoot), [Gentoo Wiki](https://wiki.gentoo.org/wiki/Flicker_Free_Boot)). Hopefully since this is an emulated text mode, it'll just be drawn on top of the native panel resolution, but it'd be good to know if there's any visible flicker/mode change delay on firmware that actually supports this mode properly.

Reply:
Is this more readable?

```
     boot.loader.systemd-boot.consoleMode = mkDefault "full";
```

Reply:
I found an unreadable bootloader worse than a flicker-free boot. Especially since NixOS contains multiple entries a User wants to choose from.

Reply:
I tried all the modes and `console-mode=1` is the only one with a reasonable font size (not too big but still readable). Yes it causes flicker but from an accessibility point of view the default configuration is not acceptable.

Reply:
`max` gives a reasonable text size at the expense of blurry font scaling and, yes, two actual panel mode changes during boot, which slows the boot process down a bit and results in a much less seamless boot transition. I feel like the kind of people who buy HiDPI laptops are generally also the kind of people likely to be unhappy about that, but the accessibility trade-off is unfortunate. (Personally I don't think the generation identifier text really adds that much since it's usually just an incrementing counter.)

My main takeaway here is that systemd-boot isn't a great option for HiDPI if you're not a fan of squinting, and the UEFI text output protocol leaves a lot to be desired in 2020... It's unfortunate that this is systemd-boot upstream's [chosen approach to HiDPI](https://github.com/systemd/systemd/issues/2750) given the amount of Linux ecosystem work towards a faster seamless boot in tandem with UEFI and modern graphics stacks.

Reply:
Until they come with their own font i don't see any better solution here.